src/main/java/org/ethereum/beacon/discovery/task/DiscoveryTaskManager.java

```diff
@@ -229,9 +230,11 @@ private void liveCheckTask() {
   }
 
   public CompletableFuture<Void> searchForNewPeers() {
-    return new RecursiveLookupTask(
-            nodeTable, this::findNodes, RECURSIVE_SEARCH_QUERY_LIMIT, Bytes32.random())
-        .execute();
+    final RecursiveLookupTask task =
+        new RecursiveLookupTask(
+            nodeTable, this::findNodes, RECURSIVE_SEARCH_QUERY_LIMIT, Bytes32.random());
+    // We wind up with a CompletableFuture<CompletableFuture> so unwrap one level.
+    return scheduler.execute(task::execute).thenCompose(Function.identity());
```

Comment:
Why move this into the scheduler?  Looks like `searchForPeers` is already running inside of the scheduler most of the time: https://github.com/PegaSysEng/discovery/blob/0e8b14fe55d983dfe845dddad91b4ce940b20b0a/src/main/java/org/ethereum/beacon/discovery/task/DiscoveryTaskManager.java#L155-L159

Maybe we need an internal method that gets scheduled at a fixed rate, and a separate public method that runs this in the scheduler?

Reply:
True, good spot.